core/src/test/scala/unit/kafka/coordinator/group/GroupMetadataManagerTest.scala

```diff
@@ -610,6 +610,42 @@ class GroupMetadataManagerTest {
   }
 
   @Test
+  def testLoadGroupsAndOffsetsFromDifferentSegments(): Unit = {
+    val groupMetadataTopicPartition = groupTopicPartition
+    val startOffset = 15L
+    val tp = new TopicPartition("foo", 0)
+
+
+    val logMock =  EasyMock.mock(classOf[Log])
+    EasyMock.expect(replicaManager.getLog(groupMetadataTopicPartition)).andStubReturn(Some(logMock))
+
+    val segment1Records = MemoryRecords.withRecords(startOffset, CompressionType.NONE,
+      createCommittedOffsetRecords(Map(tp -> 23L)) ++ Seq(buildStableGroupRecordWithMember("a")): _*)
+    val segment1End = expectGroupMetadataLoad(logMock, groupMetadataTopicPartition, startOffset, segment1Records)
+
+    val memberId = "b"
+    val committedOffsets = Map(tp -> 33L)
```

Comment:
It might be nice to include more partitions in the segments to help catch other regressions in the future. We could, for example, have 2 partitions in segment 1 and 3 partitions in segment 2.